compiler/rustc_codegen_llvm/src/intrinsic.rs

```diff
@@ -793,14 +793,16 @@ fn generic_simd_intrinsic(
         require_simd!(arg_tys[1], "argument");
         let v_len = arg_tys[1].simd_size(tcx);
         require!(
-            m_len == v_len,
+            m_len == v_len || (m_len == 8 && v_len < 8),
```

Comment:
Can you add a comment above the `require` that explains this? I found the juxtaposition between this and the error message below a bit confusing.